src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java

```diff
@@ -0,0 +1,101 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+
+/*
+ *  The Welsh-Powell algorithm is a graph coloring algorithm
+ *  used for coloring a graph with the minimum number of colors.
+ *  https://en.wikipedia.org/wiki/Graph_coloring
+ */
+
+public class WelshPowell {
+
+    public static class WPGraph {
+        private int numVer; // Number of vertices in the graph
+        private HashSet<Integer>[] adjLists; // Adjacency lists for the graph using HashSet
+
+        public WPGraph(int vertices) {
+            numVer = vertices;
+            adjLists = new HashSet[vertices];
+            for (int i = 0; i < vertices; i++) {
+                adjLists[i] = new HashSet<>();
+            }
+        }
+
+        public void addEdge(int nodeA, int nodeB) {
+            adjLists[nodeA].add(nodeB);
+            adjLists[nodeB].add(nodeA);
+        }
+
+        public HashSet<Integer> getAdjList(int vertex) {
+            return adjLists[vertex];
+        }
+
+        public int getNumVertices() {
+            return numVer;
+        }
+
+        // Helper method to check if an edge exists (used for testing)
+        boolean isEdgePresent(int v1, int v2) {
+            return adjLists[v1].contains(v2);
+        }
+    }
+
+    public static int[] welshPowellColoring(WPGraph graph) {
+        int numVer = graph.getNumVertices();
+        int[] colors = new int[numVer];
+        for (int i = 0; i < numVer; i++) {
+            colors[i] = -1; // Initialize all vertices as uncolored
+        }
+
+        LinkedList<Integer> sortedVertices = new LinkedList<>();
+        for (int i = 0; i < numVer; i++) {
+            sortedVertices.add(i);
+        }
+        sortedVertices.sort((v1, v2) -> graph.getAdjList(v2).size() - graph.getAdjList(v1).size());
```

Comment:
```suggestion
        Integer[] sortedVertices = IntStream.range(0, numVer).boxed().sorted(Comparator.comparingInt(v -> - graph.getAdjList(v).size())).toArray(Integer[] ::new);
```

Make sure to
```java
import java.util.Comparator;
import java.util.stream.IntStream;
```

I think after this change, there is no need to:
```java
import java.util.LinkedList;
```

As above: I would suggest to add a `private` method `getSortedNodes`.